doc/installing.md

```diff
@@ -0,0 +1,190 @@
+# Installing from Source (without Nix)
+
+## Prerequisites
+
+This guide assumes you have the following tools:
+
+ * [Git](https://git-scm.com/download)
+ * [Cardano Node](https://github.com/input-output-hk/cardano-node/blob/master/doc/getting-started/install.md)
+ * [Postgres Development Libraries (libpq)](https://www.postgresql.org/download/)
+ 
+In addition, Cardano DB Sync requires the following software (instructions below:
+
+ * [GHC](https://www.haskell.org/ghcup/install/) >= 8.10.7
+ * [Cabal](https://www.haskell.org/ghcup/install/) >= 3.10.1.0
+ * [libsodium-vrf](https://github.com/input-output-hk/libsodium)
+ * [secp256k1](https://github.com/bitcoin-core/secp256k1)
+
+Create a working directory for your builds:
+
+```bash
+mkdir -p ~/src
+cd ~/src
+```
+
+## Install Dependencies
+
+### Install GHC and Cabal
+
+The recommended way to install the Haskell tools is via
+[GHCup](https://www.haskell.org/ghcup/). Check [this
+page](https://www.haskell.org/ghcup/install/) for installation instructions. Its
+installation script will guide you through the installation, and warn you about package
+dependencies.
+
+Once GHCup is installed, open a new terminal (to get an updated environment) and run:
+
+```bash
+ghcup install ghc 8.10.7
+ghcup install cabal 3.10.1.0
+ghcup set ghc 8.10.7
+ghcup set cabal 3.10.1.0
+```
+
+Check that you will use the GHCup tools (and not any other installation on the system):
+
+```bash
+which cabal
+```
+
+It should print a path of this shape: `/home/<user>/.ghcup/bin/cabal`.
+
+### Install Sodium
+
+Cardano uses a [fork](https://github.com/input-output-hk/libsodium) of `libsodium` which
+exposes some internal functions and adds some other new functions.
+
+Enter a working directory for your builds:
+
+```bash
+cd ~/src
+```
+
+Checkout libsodium:
+
+```bash
+git clone https://github.com/input-output-hk/libsodium
+cd libsodium
+git checkout dbb48cc
```

Comment:
I'm wondering if there is some way to make this future proof. What is the source of truth for this revision?